main.cc

```diff
@@ -971,7 +969,7 @@ int main(int argc, char*argv[])
       }
 
       if (flag_errors)
-	    return flag_errors;
+	    exit(flag_errors);
```

Comment:
The refactoring is basically good, but I think better would be to have these new functions return int (not void) and leave the return statements instead of turning them into exit() statements.